src/examples/simple_zkapp.web.ts

```diff
@@ -0,0 +1,131 @@
+   * @param caller the privileged account
+   */
+  @method payout(caller: PrivateKey) {
+    this.account.provedState.assertEquals(Bool(true));
+
+    // check that caller is the privileged account
+    let callerAddress = caller.toPublicKey();
+    callerAddress.assertEquals(privilegedAddress);
+
+    // assert that the caller account is new - this way, payout can only happen once
+    let callerAccountUpdate = AccountUpdate.create(callerAddress);
+    callerAccountUpdate.account.isNew.assertEquals(Bool(true));
+    // pay out half of the zkapp balance to the caller
+    let balance = this.account.balance.get();
+    this.account.balance.assertEquals(balance);
+    let halfBalance = balance.div(2);
+    this.send({ to: callerAccountUpdate, amount: halfBalance });
+
+    // emit some events
+    this.emitEvent('payoutReceiver', callerAddress);
+    this.emitEvent('payout', halfBalance);
+  }
+}
+
+let Local = Mina.LocalBlockchain({ proofsEnabled: doProofs });
+Mina.setActiveInstance(Local);
+
+// a test account that pays all the fees, and puts additional funds into the zkapp
+let { privateKey: senderKey, publicKey: sender } = Local.testAccounts[0];
+
+// the zkapp account
+let zkappKey = PrivateKey.random();
+let zkappAddress = zkappKey.toPublicKey();
+
+// a special account that is allowed to pull out half of the zkapp balance, once
+let privilegedKey = PrivateKey.fromBase58(
+  'EKEeoESE2A41YQnSht9f7mjiKpJSeZ4jnfHXYatYi8xJdYSxWBep'
+);
+let privilegedAddress = privilegedKey.toPublicKey();
+
+let initialBalance = 10_000_000_000;
+let initialState = Field(1);
+let zkapp = new SimpleZkapp(zkappAddress);
+
+if (doProofs) {
+  console.log('compile');
+  await SimpleZkapp.compile();
+}
+
+console.log('deploy');
+let tx = await Mina.transaction(sender, () => {
+  let senderUpdate = AccountUpdate.fundNewAccount(sender);
+  senderUpdate.send({ to: zkappAddress, amount: initialBalance });
+  zkapp.deploy({ zkappKey });
+});
+await tx.prove();
+await tx.sign([senderKey]).send();
+
+console.log('initial state: ' + zkapp.x.get());
+console.log(`initial balance: ${zkapp.account.balance.get().div(1e9)} MINA`);
+
+let account = Mina.getAccount(zkappAddress);
+console.log('account state is proved:', account.zkapp?.provedState.toBoolean());
+
+console.log('update');
+tx = await Mina.transaction(sender, () => {
+  zkapp.update(Field(3));
+});
+await tx.prove();
+await tx.sign([senderKey]).send();
+
+// pay more into the zkapp -- this doesn't need a proof
+console.log('receive');
+tx = await Mina.transaction(sender, () => {
+  let payerAccountUpdate = AccountUpdate.createSigned(sender);
+  payerAccountUpdate.send({ to: zkappAddress, amount: UInt64.from(8e9) });
+});
+await tx.sign([senderKey]).send();
+
+console.log('payout');
+tx = await Mina.transaction(sender, () => {
+  AccountUpdate.fundNewAccount(sender);
+  zkapp.payout(privilegedKey);
+});
+await tx.prove();
+await tx.sign([senderKey]).send();
+sender;
```

Comment:
well I've created versions of this file a couple of times already, and it's very useful for me to have a default example that works on the web (the other simple zkapp example doesn't, because it uses the file system to store performance timings)

I agree that it's not pretty to have many similar versions of the same file 🤔 maybe if I move the contract to its own file it's better?

Reply:
Oh no, sorry, I just meant the line on 128, specifically the `sender;` statement

Reply:
aah ok lol